test/upgrade/upgradeTest.yaml

```diff
@@ -20,6 +20,7 @@ metadata:
   labels:
     app: upgrade-test-runner
 spec:
+  backoffLimit: 0
```

Comment:
Will it be helpful if retry the job a few times in case of transient errors of pod?

Reply:
I was running into issues (before adding the `PollUntilContextTimeout` to wait for Autopilot to scale up and schedule the pod to a new node), where there end up with multiple pods that were all trying to install Agones, and failing because there would be a previous install stuck from an old pod that had errored out. In theory this would eventually sort itself out, as the Helm command has the `atomic` flag, so it _shoud_ roll back after 10 min of failing to install, but it would seem to get stuck and not automatically roll back, and I would have to manually delete the helm release from the cluster. This might be OK to retry now, but we may still run into helm issues.